contrib/randomDictBuilder/Makefile

```diff
@@ -0,0 +1,51 @@
+PROGRAM_FILES := ../../programs/fileio.c
+
+TEST_INPUT := ../../lib
+TEST_OUTPUT := randomDict
+ARG :=
+
+all: main run clean
+
+.PHONY: test
+test: main testrun testshell clean
+
+.PHONY: run
+run:
+	echo "Building a random dictionary with given arguments"
+	./main $(ARG)
+
+main: main.o io.o random.o libzstd.a
+	gcc main.o io.o random.o libzstd.a -o main
+
+main.o: main.c
+	gcc -c main.c -I io.h -I random.h -I ../../programs -I ../../lib/common -I ../../lib -I ../../lib/dictBuilder
```

Comment:
You don't need to pass `-I` for header files in the same directory, only in different directories, and you should put it in a variable.

The Makefile doesn't need to be 100% standard, but you should use the `$(CC)` variable and `$(CFLAGS)` variables like here. This also adds compilation with `-O3` which should speed it up considerably.

```
CC ?= gcc
CFLAGS ?= -O3
INCLUDES := -I ../../programs -I ../../lib/common -I ../../lib -I ../../lib/dictBuilder

main.o: main.c
	$(CC) $(CFLAGS) $(INCLUDES) -c main.c
```